src/lib/storage/chunk_encoder.cpp

```diff
@@ -92,22 +93,41 @@ std::shared_ptr<BaseSegment> ChunkEncoder::encode_segment(const std::shared_ptr<
 }
 
 void ChunkEncoder::encode_chunk(const std::shared_ptr<Chunk>& chunk, const std::vector<DataType>& column_data_types,
-                                const ChunkEncodingSpec& chunk_encoding_spec) {
-  Assert((column_data_types.size() == chunk->column_count()),
-         "Number of column types must match the chunk’s column count.");
-  Assert((chunk_encoding_spec.size() == chunk->column_count()),
+                                const ChunkEncodingSpec& chunk_encoding_spec, const bool encode_in_parallel) {
+  const auto column_count = chunk->column_count();
+  Assert((column_data_types.size() == column_count), "Number of column types must match the chunk’s column count.");
+  Assert((chunk_encoding_spec.size() == column_count),
          "Number of column encoding specs must match the chunk’s column count.");
 
-  for (ColumnID column_id{0}; column_id < chunk->column_count(); ++column_id) {
-    const auto spec = chunk_encoding_spec[column_id];
+  auto next_column_id = std::atomic_ushort{0};
+  auto threads = std::vector<std::thread>{};
+  const auto thread_count = std::min(column_count, static_cast<uint16_t>(std::thread::hardware_concurrency()));
```

Comment:
I'm already unhappy about non-scheduler multithreading being in benchmarklib/benchmark_table_encoder.cpp, but I don't think that it should make its way into src/lib.

Reply:
Hmm ... good point. Any ideas how to get around this "performance issue" in case there is no scheduler?

Reply:
Set/unset the scheduler

Reply:
But then we have a component in the system that potentially changes the scheduler during runtime of Hyrise. Which sounds super bad to be.

Reply:
If you do it in the synthetic table generator, it is not really a component inside Hyrise.

Reply:
Yeah, but when looking at the calibration, we’ll use the synth generator and can also set a scheduler or not. Having a component fiddling with the scheduler before the actual calibration sounds like something that can easily go wrong (or am I too pessimistic?).

Reply:
If you make sure that you reset the scheduler once you are done, I don't see anything going wrong